src/systems/ec/implementations/react/packages/checkbox/Checkbox.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Icon from '@ecl/ec-react-component-icon/Icon';
+
+const Checkbox = ({
+  id,
+  checked,
+  disabled,
+  helperText,
+  invalid,
+  invalidIconLabel,
+  invalidText,
+  label,
+  hideLabel,
+  name,
+  value,
+  className,
+  ...props
+}) => {
+  const classNames = classnames(className, 'ecl-checkbox', {
+    'ecl-checkbox--invalid': invalid,
+  });
+
+  return (
+    <div className="ecl-form-group ecl-form-group--checkbox">
+      <div className={classNames}>
+        <input
+          {...props}
+          type="checkbox"
+          id={id}
+          defaultChecked={checked}
+          name={name || undefined}
+          className="ecl-checkbox__input"
+          disabled={disabled}
+        />
+        <span className="ecl-checkbox__icon" />
+        <span className="ecl-checkbox__icon ecl-checkbox__icon--checked">
+          <Icon shape="ui--check" size="xs" />
+        </span>
+
+        {label && (
+          <label
+            className={classnames(
+              'ecl-form-label ecl-form-label--inline ecl-checkbox__label',
+              {
+                'ecl-form-label--invalid': invalid,
+                'ecl-form-label--hidden': hideLabel,
+              }
+            )}
+            htmlFor={id}
+          >
+            {label}
+          </label>
+        )}
+      </div>
+
+      {invalid && invalidText && (
+        <div className="ecl-feedback-message">{invalidText}</div>
+      )}
+      {helperText && <div className="ecl-help-block">{helperText}</div>}
+    </div>
+  );
+};
+
+Checkbox.propTypes = {
+  id: PropTypes.string.isRequired,
+  checked: PropTypes.bool,
```

Comment:
It's good that you follow the [uncontrolled components](https://reactjs.org/docs/uncontrolled-components.html) practice in the context of ECL. Since we don't want to impose React, it makes sense.
But `checked` controls the `defaultChecked` property of the checkbox, thus I'd rename it `defaultChecked` as well. `checked` is misleading, as a dev I expected it to control the `checked` property... Can you please rename it?